src/components/TaskStatusIconLegend/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import IconButton from "@leafygreen-ui/icon-button";
+import Popover from "@leafygreen-ui/popover";
+import { spacing } from "@leafygreen-ui/tokens";
+import { Disclaimer, Overline } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { PopoverContainer } from "components/styles/Popover";
+import { groupedIconStatuses } from "components/TaskStatusIcon";
+import { taskStatusToCopy } from "constants/task";
+import { size } from "constants/tokens";
+
+export const LegendContent = () => (
+  <>
+    {/* @ts-expect-error */}
+    <LegendTitle>NEW ICONS LEGEND</LegendTitle>
+    <Container>
+      {groupedIconStatuses.map(({ icon, statuses }) => {
+        const label = statuses.map((status) => (
+          <Disclaimer key={status}>{taskStatusToCopy[status]}</Disclaimer>
+        ));
+        return (
+          <Row key={statuses.join()}>
+            {icon}
+            <LabelContainer>{label}</LabelContainer>
+          </Row>
+        );
+      })}
+    </Container>
+  </>
+);
+
+export const TaskStatusIconLegend: React.FC = () => {
+  const [isActive, setIsActive] = useState(false);
+
+  return (
+    <IconButton
+      onClick={() => {
+        setIsActive(!isActive);
+      }}
+      aria-label="Task Status Icon Legend"
+    >
+      <StyledIcon glyph="QuestionMarkWithCircle" />
+      <Popover
+        align="top"
+        justify="end"
+        active={isActive}
+        usePortal
+        spacing={spacing[2]}
+      >
+        <PopoverContainer>
+          <LegendContent />
+        </PopoverContainer>
+      </Popover>
+    </IconButton>
```

Comment:
I think it might be beneficial to provide some other way of closing this `Popover` besides clicking again on the `IconButton`. Since there's no `X` button, I kind of expected it to close when I clicked outside, and it took me a moment to figure out how to close it

Reply:
I added an X button on the popover. I think some users may benefit from having the popover visible while interacting with the page so I didn't include the functionality to close on outside click